### ISSUE TITLE
using grafana api key

### ISSUE THREAD

**Issue by t3h2mas** [utkarshcmu/wizzy#128]
I set up grafana w/ google login, so I'm trying to use an API key to connect wizzy to the dashboards.

I see that `'grafana:api_key'` is an entry in `validConfigs` from `config.js` L:21.

Is there a way to use an api key for authentication? Could it be documented if so?

**Comment by john-arvid** [comment]
Thank you for bringing attention to this.
Created a api key in grafana with editor role and copied the api key

To get connected I only used:
wizzy set grafana url http://grafanaip:3000
wizzy set grafana api_key averylongapikey

**Comment by Sytten** [comment]
This is indeed possible, I will add the documentation

**Comment by dideler** [comment]
@Sytten could you point me to the documentation for token based auth? I can't find it.

I'm wondering what scopes the token needs to do various imports. In this case, the scope options are the roles viewer, editor, and admin.

**Comment by dideler** [comment]
Tried all the scopes and this is the result.

| Token scope | Import dashboards | Import datasources | Import orgs |
| --- | --- | --- | --- |
| viewer | yes | no | no |
| editor | yes | no | no |
| admin | yes | yes | no |

**Comment by Sytten** [comment]
@dideler Thanks for the report, we need to fix the documentation (https://github.com/grafana-wizzy/wizzy-site). If you want to do it, I would appreciate!

**Comment by jfreeland** [comment]
> Ha yes, the commands ending in `s` are currently broken (I will try to push a fix the repo this week). Basically the authentication is not passed correctly. You should be able to do it one by one. I am still waiting on npm permissions to push a new version though.

Did this change get pushed?  I'm seeing issues with response.statusCode undefined.

**Comment by Sytten** [comment]
Its pushed on master yes, not on npm

**Comment by ttownsend78** [comment]
I think I have the latest build, but getting the below error trying to list dashboards. My Grafana instance is using generic Oauth for SSO.

```
[root@ip-10-0-8-250 grafana]# wizzy conf
Output:
{
  "grafana": {
    "envs": {
      "staging": {
        "url": "https://dashboard-dev.url.com",
        "api_key": "long api key"
      }
    }
  },
  "context": {
    "grafana": "staging"
  }
}
[root@ip-10-0-8-250 grafana]# wizzy list dashboards
Output:
No Grafana API response
Error: unable to verify the first certificate
✘ Dashboards list display failed
```

**Comment by Sytten** [comment]
It seems like you have a problem with https certificates.

**Comment by jfreeland** [comment]
> Its pushed on master yes, not on npm

Will there be a new mpm release soon?  It looks like there hasn't been a release in quite some time?

**Comment by Sytten** [comment]
Yeah I am caught up in a lot of stuff currently, I want to make one next weekend.

**Comment by nunofernandes** [comment]
Did you had the time to look at this? I did an install from npm and have the same issue when trying to export to grafana.

**Comment by jfreeland** [comment]
Would love to see a new release to npm soon. :)

**Comment by jfreeland** [comment]
Would love to see a new release soon.

**Comment by Sytten** [comment]
New release is out, let me know if you still have issues. I will publish an update to the documentation shortly.

**Comment by Tikam02** [comment]
Problem in new release -  
Grafana Version - v7.0.3
wizzy version - 0.6.0
```bash
$ wizzy list dashboards                                                                                     
Output:
Grafana API response status code = 401
No error body from Grafana API.
✘ Dashboards list display failed
```

**Comment by anthosz** [comment]
Hi,

I confirm that it doesn't works:
$ npm install -g wizzy
$ wizzy set grafana url http://.....
$ wizzy set grafana api_key key (with admin rights)
$ wizzy import dashboards
Importing 1 dashboards:
✔ Dashboard status imported successfully.
✔ 1 dashboards imported successfully.

Only public dashboard is imported..

Best regards,
